sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -92,67 +94,282 @@ public HttpPipeline getHttpPipeline() {
         return withContext(context -> createDigitalTwinWithResponse(digitalTwinId, digitalTwin, context));
     }
 
+    Mono<DigitalTwinsResponse<String>> createDigitalTwinWithResponse(String digitalTwinId, String digitalTwin, Context context) {
+        return protocolLayer
+            .getDigitalTwins()
+            .addWithResponseAsync(digitalTwinId, digitalTwin, context)
+            .flatMap(response -> {
+                try {
+                    String jsonResponse = mapper.writeValueAsString(response.getValue());
+                    DigitalTwinsResponseHeaders twinHeaders = mapper.convertValue(response.getDeserializedHeaders(), DigitalTwinsResponseHeaders.class);
+                    return Mono.just(new DigitalTwinsResponse<>(response.getRequest(), response.getStatusCode(), response.getHeaders(), jsonResponse, twinHeaders));
+                } catch (JsonProcessingException e) {
+                    return Mono.error(e);
```

Comment:
should we also log here?

Reply:
Sure, I added a log statement around these Mono.error returning code paths.